src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -2144,31 +2166,54 @@ internal void AddMemberLookupSymbolsInfo(LookupSymbolsInfo result, NamespaceOrTy
             }
         }
 
-        private void AddMemberLookupSymbolsInfoInType(LookupSymbolsInfo result, TypeSymbol type, LookupOptions options, Binder originalBinder)
+#nullable enable
+        /// <param name="accessThroughType">When collecting symbols from an extension type, we include the members from the extended type,
+        /// but those will be accessed through the extension type</param>
+        private void AddMemberLookupSymbolsInfoInType(LookupSymbolsInfo result, TypeSymbol type, LookupOptions options, Binder originalBinder, TypeSymbol? accessThroughType = null)
         {
+            Debug.Assert(accessThroughType is null || accessThroughType.IsExtension);
+
             switch (type.TypeKind)
             {
                 case TypeKind.TypeParameter:
+                    // PROTOTYPE(static) need to confirm what we want for extension members on type parameters or values of type parameters
+                    Debug.Assert(accessThroughType is null);
                     this.AddMemberLookupSymbolsInfoInTypeParameter(result, (TypeParameterSymbol)type, options, originalBinder);
                     break;
 
                 case TypeKind.Interface:
-                    this.AddMemberLookupSymbolsInfoInInterface(result, type, options, originalBinder, type);
+                    this.AddMemberLookupSymbolsInfoInInterface(result, type, options, originalBinder, accessThroughType: accessThroughType ?? type);
                     break;
 
-                // PROTOTYPE implement Lookup APIs for extension types. See TestLookupSymbolNamesInInterface and TestLookupSymbolsInInterface
-
                 case TypeKind.Class:
                 case TypeKind.Struct:
                 case TypeKind.Enum:
                 case TypeKind.Delegate:
                 case TypeKind.Array:
                 case TypeKind.Dynamic:
                 case TypeKind.Submission:
-                    this.AddMemberLookupSymbolsInfoInClass(result, type, options, originalBinder, type);
+                    this.AddMemberLookupSymbolsInfoInClass(result, type, options, originalBinder, accessThroughType: accessThroughType ?? type);
+                    break;
+
+                case TypeKind.Extension:
+                    Debug.Assert(accessThroughType is null);
+                    addMemberLookupSymbolsInfoInExtension(result, type, options, originalBinder);
                     break;
             }
+
+            return;
+
+            void addMemberLookupSymbolsInfoInExtension(LookupSymbolsInfo result, TypeSymbol type, LookupOptions options, Binder originalBinder)
+            {
+                AddMemberLookupSymbolsInfoWithoutInheritance(result, type, options, originalBinder, accessThroughType: null);
```

Comment:
>null [](http://example.com/codeflow?start=119&length=4)

Even if the difference is unobservable right now, I think we were passing the right value, given the meaning of the parameter